### PR TITLE
[llvm-lit][test] Precommit tests for curly braces in lit internal shell

### PR DESCRIPTION
**Pull request by connieyzhu**
This patch creates tests to check lit's failure to execute curly braces {}. This is a prequisite patch to https://github.com/llvm/llvm-project/pull/102830.

### CODE REVIEW

**Comment by connieyzhu** [comment]
CC: @ilovepi @petrhosek @jh7370 

Hi there, I'm writing some tests for curly braces in lit's internal shell before I implement the functionality in https://github.com/llvm/llvm-project/pull/102830. I wasn't sure if I should write more in-depth tests in this patch, since the same syntax error will be given for every test at the moment, or if I should just include more tests in my later patch once I have implemented the proper functionality.

**Comment by ilovepi** [comment]
There's some value in showing that each test used to fail, but now passes. I'm not sure it needs to be particularly exhaustive, but whatever basic tests you're considering are probably reasonable to include in this patch. Typically its nice if none of the RUN lines change, but you can easily spot a difference in the test behavior. I know I appreciate that in the middle or backend tests I review.

File: llvm/utils/lit/tests/Inputs/shtest-shell/curly-brace.txt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+## Test two commands inside curly brace.
+# RUN: { echo foo; echo bar; } | FileCheck --check-prefix=TWO-CMDS %s
+
+# TWO-CMDS: foo
+# TWO-CMDS: bar
+
+## Test nested curly brace.
+# RUN: { echo foo; { echo bar; echo baz; }; } | FileCheck --check-prefix=NESTED %s
+# RUN: { { echo foo; echo bar; }; echo baz; } | FileCheck --check-prefix=NESTED %s
```
Comment (ilovepi) [inline llvm/utils/lit/tests/Inputs/shtest-shell/curly-brace.txt:16]: I'm not sure this is checking things well enough. The nested `{...}` probably needs to result in something that's distinct from the regular case. As written, you cant distinguish the nested version from the un-nested version. Maybe you can use a `|` or file redirection + `cat`?

File: llvm/utils/lit/tests/shtest-shell.py [hunk 2]
```diff
@@ -4,7 +4,7 @@
 # FIXME: Temporarily dump test output so we can debug failing tests on
 # buildbots.
 # RUN: cat %t.out
-# RUN: FileCheck --input-file %t.out %s
+# RUN: FileCheck --input-file %t.out %s --dump-input-context=1000
```
Comment (ilovepi) [inline llvm/utils/lit/tests/shtest-shell.py:7]: leftover from debugging?

Comment (connieyzhu) [reply llvm/utils/lit/tests/shtest-shell.py:7]: Yes, forgot to delete that. I will remove it in my next change.

File: llvm/utils/lit/tests/Inputs/shtest-shell/curly-brace.txt [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+# ONE-CMD: bar
+
+## Test two commands inside curly brace.
+# RUN: { echo foo; echo bar; } | FileCheck --check-prefix=TWO-CMDS %s
+
+# TWO-CMDS: foo
+# TWO-CMDS: bar
+
+## Test nested curly brace.
+# RUN: { echo foo > %t/temp.txt; { echo bar; echo baz; }; cat %t/temp.txt; } | FileCheck --check-prefix=NESTED %s
```
Comment (ilovepi) [inline llvm/utils/lit/tests/Inputs/shtest-shell/curly-brace.txt:15]: I don't see a `mkdir %t`, so I'm not sure this works as intended. If you don't need other files, you can just use `%t`.

File: llvm/utils/lit/tests/Inputs/shtest-shell/curly-brace.txt [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+# NESTED: bar
+# NESTED: baz
+# NESTED: foo
```
Comment (ilovepi) [inline llvm/utils/lit/tests/Inputs/shtest-shell/curly-brace.txt:19]:
```suggestion
#      NESTED: bar
# NESTED-NEXT: baz
# NESTED-NEXT: foo
```

**Comment by jh7370** [comment]
I'm not opposed to this PR being developed, but I don't want it and the follow-on PR landed without the RFC discussion I've requested in the source issue: as noted there, I'm not convinced this functionality is needed in lit (not going to elaborate further here, to avoid splitting the discussion).

**Pull request closed without merging**